extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -274,13 +339,77 @@ oppia.directive('imageWithRegionsEditor', [
             $scope.hoveredRegion = index;
           }
         };
+        $scope.onMouseMoveRegion = function() {
+          if ($scope.userIsCurrentlyDragging ||
+              $scope.userIsCurrentlyResizing) {
+            return;
+          }
+          region = reg = cornerAndDimensionsFromRegionArea(
+            $scope.$parent.value.labeledRegions[
+              $scope.hoveredRegion].region.area);
+          if (!$scope.xDirectionToggled && !$scope.yDirectionToggled) {
+            if ($scope.mouseY <= region.y + $scope.resizeRegionBorder &&
+                $scope.mouseX >= region.width + region.x -
+                $scope.resizeRegionBorder) {
+              // Along top right
+              $scope.xDirection = -1;
```

Comment:
Refactored the conditions so that the xDirection and yDirection are set independently.